tests/services/profile.service.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { Profile } from '../../src/app/core/models/profile.model';
+import { ProfileService } from '../../src/app/core/services/profile.service';
+
+let profileService: any;
+
+const EMPTY_PROFILES = [];
+const VALID_PROFILES = [
+  <Profile>{
+    profileName: 'CATcher',
+    encodedText: 'CATcher-org/public_data'
+  }
+];
+const INVALID_PROFILES = [
+  {
+    profileName: 'CATcher'
+  }
+];
+
+/**
+ * Stub to mock the fetch API Call which retrieves the profiles
+ * from the external repository.
+ */
+class PromiseStub extends Promise<Response> {
```

Comment:
@anubh-v @dingyuchen @ptvrajsk 

As of now, I have decided to use a `Stub` to mimic the `Promise` returned from the retrieval of external profiles. Do let me know of your thoughts from this design ><

Reply:
This is also due to the fact that it is **not possible** to mimic the `fetch` return call using spy objects.

Reply:
I think it's good :+1: !

Reply:
How about this instead? :
```
      const mockResponse = { json: () => Promise.resolve({profiles: VALID_PROFILES}) } as Response;
      spyOn(window, 'fetch').and.returnValue(Promise.resolve(mockResponse));
```

Reply:
Yes this was what I was looking for, thanks @anubh-v !